### DISCUSSION TITLE
weblate.discovery.discovery

### DISCUSSION THREAD

**Discussion by tomkolp** [WeblateOrg/weblate#5998] [General] [answered]
After importing from vcs, the process starts and takes longer than the import process itself. What exactly does this process do?

**Comment by nijel** [comment, marked as answer]
See https://docs.weblate.org/en/latest/admin/addons.html#component-discovery

**Reply by tomkolp** [reply]
I completely forgot about it. The addon works fine, but the "crawlers" make importing vcs so long. There are just so many inquiries the celer has gotten overbooked.
![image](https://user-images.githubusercontent.com/39633242/117962913-0f2f1000-b320-11eb-914a-49fb12048240.png)

**Reply by nijel** [reply]
Are you sure it's related to crawlers? The unauthenticated pages should not trigger background tasks (with exception of repo notification hooks, if enabled).

You can check the log to see what tasks are being processed, look for `Received task:` entries.

**Reply by tomkolp** [reply]
Unfortunately, the dish does not fully understand the logs yet.
![image](https://user-images.githubusercontent.com/39633242/117977176-d77c9400-b330-11eb-9986-d6577021bb40.png)

**Reply by nijel** [reply]
This one is for the notify queue :-)

**Reply by tomkolp** [reply]
Yesterday at 19:00 I uploaded the latest version of weblate. After rebooting, the celer quickly started filling up again. The whole Weblate works fine, I don't see a big drop in performance.
![image](https://user-images.githubusercontent.com/39633242/118085892-6b973b80-b3c3-11eb-8bf7-0717efb0a5a3.png)
![image](https://user-images.githubusercontent.com/39633242/118084133-74d2d900-b3c0-11eb-8703-0352222920d6.png)
![image](https://user-images.githubusercontent.com/39633242/118084165-861be580-b3c0-11eb-98b1-d82ecfa66b08.png)
![image](https://user-images.githubusercontent.com/39633242/118089037-380ae000-b3c8-11eb-9a47-05d8e0bdd373.png)
![image](https://user-images.githubusercontent.com/39633242/118099537-a7d39780-b3d5-11eb-8144-7d413e5ea79d.png)

**Reply by nijel** [reply]
It seems that the add-ons are being executed way too frequently. Do you know when this started to happen?

**Reply by nijel** [reply]
Also, how do you trigger updating Weblate repositories from upstream? It might be related to this...

**Reply by tomkolp** [reply]
First time yesterday.  The last successful import was two weeks ago.  I upload bleeding updates up to two days from release.

**Reply by tomkolp** [reply]
> Also, how do you trigger updating Weblate repositories from upstream? It might be related to this...

Pull from the repository menu

**Reply by nijel** [reply]
I'm curious if 17cd3369626bb26ac261278cceb760f1b6a270c4 will make the difference in this case (it should not, but the new code is less fragile).

**Reply by tomkolp** [reply]
> I'm curious if [17cd336](https://github.com/WeblateOrg/weblate/commit/17cd3369626bb26ac261278cceb760f1b6a270c4) will make the difference in this case (it should not, but the new code is less fragile).

Everything is back to normal after the update. I will mark the topic as resolved after re-importing from vcs.

**Reply by tomkolp** [reply]
After upgrading from vcs, the celer will not overflow. There is an update stamp in the added components all the time, this was not the case before. Once a component was added, it was practically ready immediately. Now after adding a component, I have to wait for 1600 others to check before the update mark disappears.
![image](https://user-images.githubusercontent.com/39633242/118244688-1da43580-b4a0-11eb-9c17-ae14f8d0887b.png)
![image](https://user-images.githubusercontent.com/39633242/118245518-382ade80-b4a1-11eb-8fc3-4b4538006ef7.png)
![image](https://user-images.githubusercontent.com/39633242/118245617-555fad00-b4a1-11eb-9609-4c6ccb20ad53.png)

**Reply by tomkolp** [reply]
Three hours from adding a component ...
![image](https://user-images.githubusercontent.com/39633242/118265773-f22e4480-b4b9-11eb-8aac-d8aeb448508b.png)
![image](https://user-images.githubusercontent.com/39633242/118265840-0d994f80-b4ba-11eb-9d51-aa6f12c7ab6c.png)
![image](https://user-images.githubusercontent.com/39633242/118265887-1c800200-b4ba-11eb-9335-114a2eeea86d.png)
![image](https://user-images.githubusercontent.com/39633242/118266054-5cdf8000-b4ba-11eb-8d59-c5b914d24609.png)

**Reply by nijel** [reply]
That's bad. I could reproduce some of that, and it should be fixed by a2a17f6cb608fc2e0fdcb79bff7388fa08e4aa28 and 2487b1c6c950ac61e406fd5895f5384f6ff13e5a.

**Comment by tomkolp** [comment]
Autobackup for borg has stopped working.  I tried to trigger the backup manually.  The task is handed over to the celer and nothing else happens.
![Screenshot_20210515-112437](https://user-images.githubusercontent.com/39633242/118355627-42b4a900-b571-11eb-958a-e4b4d9c6556b.png)
![Screenshot_20210515-112400](https://user-images.githubusercontent.com/39633242/118355641-506a2e80-b571-11eb-86e6-24931f4c19bd.png)

**Reply by tomkolp** [reply]
After the restart, 9 backup tasks were done without any problems. Looks like a celer problem. I guess it stops working properly after a while. After restart, everything returns to normal.
